dxp/architecture/rich_text_editors.md

```diff
@@ -0,0 +1,158 @@
+
+## JavaScript
+
+From **Liferay DXP 7.3** forward, the module [`frontend-editor-ckeditor-web`](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/index.js) exposes React versions of CKEditor adapted to different scenarios:
+
+| Rich Text Needs | Solution |
+| --- | --- |
+| Basic Rich Text Features (text formatting, links, images...) | Use the **`<Editor />`** React component
+| Contextual Inline Editing Experience (page-editing, blogs...) | Use the **`<BallonEditor />`** React component |
+| Default Inline Editing Experience (single fixed toolbar) | Use the **`<InlineEditor />`** React component |
+
+Please, read through our [General Guidelines](#General-Guidelines) to pick the necessary component for your needs and then use it accordingly:
+
+```javascript
+<Editor
+	autoFocus={autoFocus}
+	configurationName="comment"
+	id={id}
+	initialValue={textareaContent}
+	onChange={onTextareaChange}
+	placeholder={Liferay.Language.get(
+		'type-your-comment-here'
+	)}
+/>
+```
+
+## Java
+
+The following are the main 2 Java interfaces to know when dealing with Rich Text Editor in DXP
+
+### [`EditorRenderer`](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/modules/apps/frontend-editor/frontend-editor-api/src/main/java/com/liferay/frontend/editor/EditorRenderer.java#L20-L32)
+
+> Defines an available Rich Text Editor
+
+Implement `EditorRenderer` in the unlikely event that you need to create a new Rich Text Editor.
+
+```java
+@Component(
+	property = "name=ckeditor", service = {Editor.class, EditorRenderer.class}
+)
+public class CKEditorEditor implements Editor, EditorRenderer {
+	[...]
+}
+```
+
+You can read more about this API in the Rich Text Editor [Anatomy](#Anatomy) section.
+
+### [`EditorConfigContributor`](https://github.com/liferay/liferay-portal/blob/61601e89b64240db742eceaf82e86460620bcd97/portal-kernel/src/com/liferay/portal/kernel/editor/configuration/EditorConfigContributor.java#L105-L130)
+
+> Provides an interface for setting the editor's configuration
+
+Currently, you can implement `EditorConfigContributor` to [Modify an Editor’s Configuration](https://portal.liferay.dev/docs/7-0/tutorials/-/knowledge_base/t/modifying-an-editors-configuration).
+
+```java
+@Component(
+    property = {"editor.name=ckeditor"},
+    service = EditorConfigContributor.class
+)
+public class MyEditorConfigContributor extends BaseEditorConfigContributor {
+
+	@Override
+	public void populateConfigJSONObject(JSONObject jsonObject) {
+		// Modifies `jsonObject` in place for the desired configuration
+	}
+}
+```
+
+> **Warning:** Keep in mind that Editor Config Contributors stack one on top of each other following a specificity algorithm. All Config Contributors that apply to a given editor will run over the `jsonObject` object mutating it in place in order.
```

Comment:
"one on top" → "on top"

and:

"in place in order" → "in place, in order" might be clearer